tools/build.sh

```diff
@@ -27,32 +27,61 @@ releaseDir="$baseDir/release/$modname"
 mkdir -p "$releaseDir"
 cp -r "${baseDir}/addons" "${releaseDir}/"
 cp -r "${baseDir}/optionals" "${releaseDir}/"
-cp "${baseDir}"/*.paa "${baseDir}"/*.cpp "${releaseDir}/"
+cp "${baseDir}"/*.paa "${baseDir}"/*.cpp "${baseDir}/README.md" "${releaseDir}/"
 
-build_pbos() {
-	local addonsdir="$releaseDir/$1"
-	find "$addonsdir" -maxdepth 1 ! -path "$addonsdir" -type d | while read component; do
-		echo "packing $component"
 
-		componentname=`basename "$component"`
-		componentpath=`dirname "$component"`
+build_pbo() {
+	componentname=`basename "${1}"`
+	componentpath=`dirname "${1}"`
+	echo "packing $componentname"
 
-		pbofilename="${pboprefix}${componentname}.pbo"
-		pbofilepath="${componentpath}/$pbofilename"
+	pbofilename="${pboprefix}${componentname}.pbo"
+	pbofilepath="${componentpath}/$pbofilename"
 
-		"${armakePath}" build -f -p "$component" "$pbofilepath"
-		rm -r "$component"
+	"${armakePath}" build -f -p "${1}" "$pbofilepath"
 
-		if [[ ! -f "$pbofilepath" ]]; then
-			echo "failed"
-			exit 2
-		fi
+	if [[ ! -f "$pbofilepath" ]]; then
+		echo "failed"
+		exit 2
+	fi
+}
+
+merge_readme() {
+	componentname=`basename "${1}"`
+	componentpath=`dirname "${1}"`
+	echo "merging $componentname readme"
+
+	sed -i '$a ***' "${releaseDir}/README.md"
+	cat "$componentpath/$componentname/README.md" >> "$releaseDir/README.md"
+}
+
+pack_directory() {
+	find "$1" -maxdepth 1 ! -path "$1" -type d | while read component; do
+		merge_readme "${component}"
+		build_pbo "${component}"
+		rm -r "${component}"
 	done
 }
 
-# pbo and remove folders in addons directory
-build_pbos addons
-build_pbos optionals
+#format readme
+readmeFile="${releaseDir}/README.md"
+
+sed -i '4d' "${readmeFile}"
+sed -i '7,$d' "${readmeFile}"
+sed -i '$a ***\n***' "${readmeFile}"
+sed -i '$a ## Components' "${readmeFile}"
+sed -i '$a These components are part of Gruppe Adler Mod.' "${readmeFile}"
+pack_directory "$releaseDir/addons"
+
+sed -i '$a ***\n***' "${readmeFile}"
+sed -i '$a ## Optional Components' "${readmeFile}"
+sed -i '$a These components are are whitelisted on our servers. You can activate a component by moving its *.pbo file from *the optionals* to the addons *directory*.' "${readmeFile}"
+pack_directory "$releaseDir/optionals"
+
+npm install -g markdown-pdf
```

Comment:
`-g` can lead to permissions problems on some systems – it'd be a bit safer to not use `-g` and use the binary from `node_modules/.bin/markdown-pdf`

… but if it works with Travis it's fine with me. 
Maybe I should fix my system permissions instead so -g is not a problem :laughing:

Reply:
I haven't actually tested it with Travis. So without -g it would be:
```
npm install markdown-pdf
node_modules/.bin/markdown-pdf "${readmeFile}"
rm -r "node_modules"
```

... or do we add `tools/node_modules` to .gitignore instead of removing it?

Reply:
right. having `node_modules` in the project seems weird. idk. 

let's try it with -g first, as it's the least complicated variant.